#! /bin/sh /usr/share/dpatch/dpatch-run
## 40_sockaddr_ll.dpatch by Dean Gaudet <dean@arctic.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: From Dean Gaudet <dean@arctic.org>: handle AF_PACKET addresses in
## DP: pcap_findalldevs.

@DPATCH@
diff -urNad libpcap0.8-0.8.3/fad-getad.c /tmp/dpep.HnC1yl/libpcap0.8-0.8.3/fad-getad.c
--- libpcap0.8-0.8.3/fad-getad.c	2004-03-12 00:04:52.000000000 +0100
+++ /tmp/dpep.HnC1yl/libpcap0.8-0.8.3/fad-getad.c	2004-11-07 22:21:16.000000000 +0100
@@ -58,6 +58,10 @@
 #include "os-proto.h"
 #endif
 
+#ifdef AF_PACKET
+# include <linux/if_packet.h>
+#endif
+
 /*
  * This is fun.
  *
@@ -101,6 +105,11 @@
 		return (sizeof (struct sockaddr_in6));
 #endif
 
+#ifdef AF_PACKET
+	case AF_PACKET:
+		return (sizeof (struct sockaddr_ll));
+#endif
+
 	default:
 		return (sizeof (struct sockaddr));
 	}
